geopolars/src/geoseries.rs

```diff
@@ -307,6 +310,60 @@ impl GeoSeries for Series {
         Ok(series)
     }
 
+    fn explode_geometries(&self) -> Result<Series> {
+        let mut nested_vector = vec![];
+
+        for geometry in iter_geom(self) {
+            match geometry {
+                Geometry::Point(geometry) => {
+                    let point = vec![Geometry::Point(geometry)];
+                    nested_vector.push(point)
+                }
+                Geometry::MultiPoint(geometry) => {
+                    let points: Vec<Geometry> = geometry.into_iter().map(Geometry::Point).collect();
+                    nested_vector.push(points)
```

Comment:
Instead of having `nested_vector` being `vec![vec![geometry]]`, could we push individual parts into that function?
```rs
for geom in geometry.into_iter() {
    output_vector.push(Geometry::Point(geom))
}
```

That means
1. we avoid the extra vec construction for each geometry
2. we avoid an extra iteration down below when you flatten

Reply:
Addressed in https://github.com/geopolars/geopolars/pull/111/commits/43f9edbe614720380e3d6b512595774a966885c5